spec/models/user_spec.rb

```diff
@@ -158,4 +155,60 @@
       end
     end
   end
+
+  describe "#send_welcome_email (after_save callback)" do
+    context "when the user does NOT have an email address"  do
+      before do
+        subject.update(email: nil)
+        subject.save!
+      end
+
+      specify "#save does NOT call #send_welcome_email" do
+        is_expected.not_to receive(:send_welcome_email)
+        subject.sent_emails.clear
+        subject.save!
+      end
+    end
+
+    context "when the user DOES have an email address"  do
+      before { subject.update(email: "some@email.address") }
+
+      specify "#save DOES call #send_welcome_email" do
+        is_expected.to receive(:send_welcome_email)
+        subject.sent_emails.clear
+        subject.save!
+      end
+
+      describe "#send_welcome_email" do
+        let(:an_email) { double(:an_email)}
+        before do
+          allow(an_email).to receive(:deliver_now)
+          allow(UserMailer).to receive(:welcome_email).and_return(an_email)
+        end
+
+        context "when NO previous welcome emails have been sent" do
+          before { subject.sent_emails.clear }
+
+          it "sends an email" do
+            expect(an_email).to receive(:deliver_now)
+            expect(UserMailer).to receive(:welcome_email).with(subject).and_return(an_email)
+            subject.save!
+          end
+
+          specify { expect { subject.save! }.to change(subject, :needs_welcome_email?).from(true).to(false) }
```

Comment:
Again Hound and rubocop should be moaning about long lines here?

Reply:
That one is not long enough to trigger the rule ... do you want to (separately, maybe) amend  the rule ?

Reply:
Nah, wait until #172 for that.